packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -87,6 +87,29 @@ describe('<TreeItem />', () => {
       .to.equal('endIcon');
   });
 
+  it('should allow conditional child', () => {
+    function TestComponent() {
+      const [hide, setState] = React.useState(false);
+
+      return (
+        <React.Fragment>
+          <button type="button" onClick={() => setState(true)}>
+            Hide
+          </button>
+          <TreeView defaultExpanded={['1']}>
+            <TreeItem nodeId="1" label="1" data-testid="1">
+              {!hide && <TreeItem nodeId="test" label="test" data-testid="2" />}
+            </TreeItem>
+          </TreeView>
+        </React.Fragment>
+      );
+    }
+    const { getByText, queryByText } = render(<TestComponent />);
+    expect(getByText('test')).to.not.be.null;
```

Comment:
Please don't use getByText. It's extremely brittle. You can leverage ByTestId since you already added testids.

Also: what is the expected aria-expanded before and after the click?

Reply:
Ok, as I said I mirrored the existing TreeView test.

Will update

Reply:
> Ok, as I said I mirrored the existing TreeView test. There are many getByText.

No worries. Sometimes these slip by.